input.go

```diff
@@ -85,3 +89,21 @@ type ContractCallInput struct {
 	// The function must be public (e.g. in Iele `define public @functionName(...)`)
 	Function string
 }
+
+// GetCodeUpgrade gets the code upgrade from the arguments
+func (callInput *ContractCallInput) GetCodeUpgrade() ([]byte, error) {
```

Comment:
On top of what @sasurobert said, this method can create confusion, because `ContractCreateInput` already contains a `ContractCode` field, and both `ContractCreateInput` and `ContractCallInput` contain a `VMInput`. It might be better to simply use `codeUpgrade := callInput.Arguments[0]` during an upgrade. I suggest removing it entirely, or creating a dedicated method in the `runtimeContext`, like `runtimeContext.GetCodeUpgradeFromArgs()`.

Reply:
Removed `GetCodeUpgrade()` and `GetCodeMetadataUpgrade()`.

*Motivation for that design:* `ContractCreateInput` and `ContractCallInput` should be high-level abstractions and encapsulate common-sense logic about the data they hold. If they fail to do so, then they have to leak their internals to client code. `Arguments` is not an "internal", but `Arguments[0]` and `Arguments[1]`, are to a certain degree. Knowledge and interpretation of data and logic around data should stay close to the actual data. The only practical reason to move this logic away from `VMCommon` is if other future Virtual Machines would fetch upgrade code and metadata differently than we currently planned for Arwen - but that would, be, actually, very bad, since we'd loose consistency to an unreasonable degree.